python/plugins/processing/algs/qgis/SplitWithLines.py

```diff
@@ -0,0 +1,175 @@
+                outFeat.setAttributes(attrsA)
+                inGeoms = [inGeom]
+                lines = spatialIndex.intersects(inGeom.boundingBox())
+
+                if len(lines) > 0:  # has intersection of bounding boxes
+                    splittingLines = []
+
+                    for i in lines:
+                        try:
+                            splitGeom = splitGeoms[i]
+                        except:
+                            continue
+
+                        # check if trying to self-intersect
+                        if sameLayer:
+                            if inFeatA.id() == i:
+                                continue
+
+                        if inGeom.intersects(splitGeom):
+                            splittingLines.append(splitGeom)
+
+                    if len(splittingLines) > 0:
+                        for splitGeom in splittingLines:
+                            splitterPList = vector.extractPoints(splitGeom)
+                            outGeoms = []
+
+                            while len(inGeoms) > 0:
+                                inGeom = inGeoms.pop()
+                                inPoints = vector.extractPoints(inGeom)
+
+                                if inGeom.intersects(splitGeom):
+                                    try:
+                                        result, newGeometries, topoTestPoints = inGeom.splitGeometry(splitterPList, False)
+                                    except:
+                                        ProcessingLog.addToLog(ProcessingLog.LOG_WARNING,
+                                                               self.tr('Geometry exception while splitting'))
+                                        result = 1
+
+                                    # splitGeometry: If there are several intersections
+                                    # between geometry and splitLine, only the first one is considered.
+                                    if result == 0:  # split occurred
+
+                                        if inPoints == vector.extractPoints(inGeom):
+                                            # bug in splitGeometry: sometimes it returns 0 but
+                                            # the geometry is unchanged
+                                            outGeoms.append(inGeom)
+                                        else:
+                                            inGeoms.append(inGeom)
+
+                                            for aNewGeom in newGeometries:
+                                                inGeoms.append(aNewGeom)
+                                    else:
+                                        outGeoms.append(inGeom)
+                                else:
+                                    outGeoms.append(inGeom)
+
+                            inGeoms = outGeoms
+
+                for aGeom in inGeoms:
+                    passed = True
+
+                    if aGeom.wkbType == 2 or aGeom.wkbType == -2147483646:
```

Comment:
At the moment this is only passing LineString or LineString25D. I'd suggest changing it to

    if QgsWkbTypes.geometryType( aGeom.wkbType() )  == QgsWkbTypes.LineGeometry and not QgsWkbTypes.isMultiType( aGeom.wkbType() ):

Then it'll support LineStringZ, LineStringM, ...ZM, curved types, etc, and be future proof in case more are added in future.